packages/compose/src/hooks/use-dialog/index.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import mergeRefs from 'react-merge-refs';
+
+/**
+ * WordPress dependencies
+ */
+import { useRef, useCallback, useEffect } from '@wordpress/element';
+import { ESCAPE } from '@wordpress/keycodes';
+
+/**
+ * Internal dependencies
+ */
+import useConstrainedTabbing from '../use-constrained-tabbing';
+import useFocusOnMount from '../use-focus-on-mount';
+import useFocusReturn from '../use-focus-return';
+import useFocusOutside from '../use-focus-outside';
+
+/**
+ * Returns a ref and props to apply to a dialog wrapper to enable the following behaviors:
+ *  - constrained tabbing.
+ *  - focus on mount.
+ *  - return focus on unmount.
+ *  - focus outside.
+ *
+ * @param {Object} options Dialog Options.
+ */
+function useDialog( options ) {
+	const onClose = useRef();
+	useEffect( () => {
+		onClose.current = options.onClose;
+	}, [ options.onClose ] );
+	const constrainedTabbingRef = useConstrainedTabbing();
+	const focusOnMountRef = useFocusOnMount();
+	const focusReturnRef = useFocusReturn();
+	const focusOutsideProps = useFocusOutside( options.onClose );
+	const closeOnEscapeRef = useCallback( ( node ) => {
+		if ( ! node ) {
+			return;
+		}
+
+		node.addEventListener( 'keydown', ( event ) => {
+			// Close on escape
+			if ( event.keyCode === ESCAPE && onClose.current ) {
+				event.stopPropagation();
+				onClose.current();
+			}
+		} );
+
+		node.addEventListener( 'mousedown', ( event ) => {
+			// I'm not really certain what this is for but it matches the previous behavior.
```

Comment:
Actually the previous comment for this was:
>   onMouseDown is disabled as this can cause interactions with other DOM elements

By testing it, it seems to interact with other elements and not trigger `onClose`. So it would be good to replace the comment here.

Reply:
I'm sorry for not being clear. I tried it yesterday and I think it didn't close when clicking outside the inserter, in another block in the editor... I tried to verify now and it was closing as well without the `mousedown` and works just fine 🤔 .